src/app/clusters/push-av-stream-transport-server/tests/TestPushAVStreamTransportCluster.cpp

```diff
@@ -0,0 +1,1162 @@
+
+    // Create TransportOptionsStruct object
+    TransportOptionsDecodableStruct transportOptions;
+    transportOptions.streamUsage = StreamUsageEnum::kAnalysis;
+    transportOptions.videoStreamID.SetValue(1);
+    transportOptions.audioStreamID.SetValue(2);
+    transportOptions.endpointID       = 1;
+    transportOptions.url              = "rtsp://192.168.1.100:554/stream";
+    transportOptions.triggerOptions   = triggerOptions;
+    transportOptions.containerOptions = containerOptions;
+    transportOptions.expiryTime.SetValue(1000);
+
+    EXPECT_EQ(logic.ValidateIncomingTransportOptions(transportOptions), Status::ConstraintError);
+}
+
+void PrintBufHex(const uint8_t * buf, size_t size)
+{
+    for (size_t i = 0; i < size; ++i)
+    {
+        printf("%02X ", buf[i]); // Print each byte as 2-digit hex
+
+        // Optional: print a newline every 16 bytes
+        if ((i + 1) % 16 == 0)
+            printf("\n");
+    }
+
+    // Add final newline if buffer didn’t end on a 16-byte boundary
+    if (size % 16 != 0)
+        printf("\n");
+}
+
+TEST_F(TestPushAVStreamTransportServerLogic, TestCurrentConnectionsAttributeAccess)
+{
+    PushAvStreamTransportServer server(1, BitFlags<Feature>(1));
+
+    // Create CMAFContainerOptionsStruct object
+    CMAFContainerOptionsStruct cmafContainerOptions;
+    cmafContainerOptions.chunkDuration = 1000;
+    cmafContainerOptions.metadataEnabled.ClearValue();
+    std::string cencKey   = "deadbeefdeadbeef";
+    std::string cencKeyID = "dadabeefdadabeef";
+    cmafContainerOptions.CENCKey.SetValue(ByteSpan(reinterpret_cast<const uint8_t *>(cencKey.c_str()), cencKey.size()));
+    cmafContainerOptions.CENCKeyID.SetValue(ByteSpan(reinterpret_cast<const uint8_t *>(cencKeyID.c_str()), cencKeyID.size()));
+
+    // Create ContainerOptionsStruct object
+    ContainerOptionsStruct containerOptions;
+    containerOptions.containerType = ContainerFormatEnum::kCmaf;
+    containerOptions.CMAFContainerOptions.SetValue(cmafContainerOptions);
+
+    // Create a TransportMotionTriggerTimeControlStruct object
+    TransportMotionTriggerTimeControlDecodableStruct motionTimeControl;
+
+    motionTimeControl.initialDuration      = 5000;
+    motionTimeControl.augmentationDuration = 2000;
+    motionTimeControl.maxDuration          = 30000;
+    motionTimeControl.blindDuration        = 1000;
+
+    TransportTriggerOptionsDecodableStruct triggerOptions;
+    triggerOptions.triggerType = TransportTriggerTypeEnum::kMotion;
+
+    // Create transport zone options structs
+    Structs::TransportZoneOptionsStruct::Type zone1;
+    zone1.zone.SetNonNull(1);
+    zone1.sensitivity.SetValue(5);
+
+    Structs::TransportZoneOptionsStruct::Type zone2;
+    zone2.zone.SetNonNull(2);
+    zone2.sensitivity.SetValue(10);
+
+    // Encode them into a TLV buffer
+    uint8_t tlvBuffer[512];
+    TLV::TLVWriter writer;
+    writer.Init(tlvBuffer, sizeof(tlvBuffer));
+
+    TLV::TLVWriter containerWriter;
```

Comment:
Could we make the logic type-safe and then test the logic instead of testing via the TLV cluster interface?
Working with TLV and raw data encoding is a lot more complex.